firehose/parsers/flawfinder.py

```diff
@@ -0,0 +1,119 @@
+#!/usr/bin/env python
+#
+#   Copyright 2017 David Carlos  <ddavidcarlos1392@gmail.com>
+#   This library is free software; you can redistribute it and/or
+#   modify it under the terms of the GNU Lesser General Public
+#   License as published by the Free Software Foundation; either
+#   version 2.1 of the License, or (at your option) any later version.
+#
+#   This library is distributed in the hope that it will be useful,
+#   but WITHOUT ANY WARRANTY; without even the implied warranty of
+#   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+#   Lesser General Public License for more details.
+#
+#   You should have received a copy of the GNU Lesser General Public
+#   License along with this library; if not, write to the Free Software
+#   Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301
+#   USA
+
+import sys
+import re
+from subprocess import check_output
+from firehose.model import Message, Point, \
+    File, Location, Generator, Metadata, Analysis, Issue
+
+
+def main():
+    """ Main entry to flawfinder parser """
+    try:
+        arg_file = sys.argv[1]
+        report = open(arg_file, 'r')
+    except IOError:
+        print("File note found")
+    except IndexError:
+        print("Missing input file")
+    analysis = parse_file(report)
+    sys.stdout.write(str(analysis.to_xml()))
+    sys.stdout.write('\n')
+
+
+def parse_file(infile):
+    """ Parser flawfinder output
+
+    :infile: file-like object
+    :returns: Firehose Analysis object, representing the final XML.
+    """
+
+    line = infile.readline()
+    generator = Generator(name='flawfinder',
+                          version=get_flawfinder_version(line))
+    metadata = Metadata(generator, None, None, None)
+    analysis = Analysis(metadata, [])
+
+    pattern = "\([a-z]*\)"
+    pattern2 = "\[[0-9]\]"
+    prog = re.compile(pattern)
+    prog2 = re.compile(pattern2)
+
+
+    while line:
+        testid = None
+        try:
+            testid = re.findall(r"\(([a-z]+)\)", line)[0]
+        except IndexError:
+            pass
+        if line != "\n" and (prog.search(line) and prog2.search(line)):
+
+            issue_cwe = ''
+            issue_path = line.split(":")[0]
+            issue_line = line.split(":")[1]
+
+            location = Location(file=File(issue_path, None),
+                                function=None,
+                                point=Point(int(issue_line), 0))
+
+            message_line = infile.readline()
+            issue_message = ""
+            while message_line != "\n" and not \
+            (prog.search(message_line) and prog2.search(message_line)):
+                issue_message += " " + message_line
+                try:
+                    cwe = re.findall("CWE-([0-9]+)", message_line)[0]
```

Comment:
I don't like using try/except for parsing; I prefer to use non-exception program flow (rationale: sometimes we end up catching a different exception than the one we thought we were, so it can conceal bugs).

Anything that matches the ``([0-9]+)`` is known to be convertible to ``int``, so how about:

    # list of ints
    cwes = []

before the "while message_line" loop, and:

    cwes.extend([int(m.group(0))
                 for m in re.findall("CWE-([0-9]+)", message_line)])

here, and then after the "message_line" loop:

    if cwes:
       first_cwe = cwes[0]
    else
       first_cwe = None

to avoid the two try/except clauses.

Please reinstate the comment about issue #35